release.sh

```diff
@@ -314,6 +317,21 @@ function sign_release() {
   if [ -z "${SIGN_IMAGES:-}" ]; then # Temporary Feature Gate
     return 0
   fi
+
+  # Notarizing mac binaries needs to be done before cosign as it changes the checksum values
+  # of the darwin binaries
+  if [ -n "${APPLE_CODESIGN_KEY}" && -n "${APPLE_CODESIGN_PASSWORD}" && -n "${APPLE_NOTARY_API_KEY}" ]; then
+    FILES=$(find * -type f -name "*darwin*")
+    for file in $FILES; do
+      rcodesign sign "${file}" --p12-file="${APPLE_CODESIGN_KEY}" \
+        --code-signature-flags=runtime \
+        --p12-password-file="${APPLE_CODESIGN_PASSWORD_FILE}"
+    done
+    zip files.zip $FILES
+    sha256sum ${ARTIFACTS_TO_PUBLISH} > checksums.txt
+    rcodesign notary-submit files.zip --api-key-path="${APPLE_NOTARY_API_KEY}"  --wait
+  fi
```

Comment:
I'm seeing multiple [shellcheck](https://www.shellcheck.net/) issues in this fragment. Please, run the code through shellchek and fix the problems.

Also, to avoid line limits, you could do:

```bash
while read -r file; do
  rcodesign sign "${file}" --p12-file="${APPLE_CODESIGN_KEY}" \
        --code-signature-flags=runtime \
        --p12-password-file="${APPLE_CODESIGN_PASSWORD_FILE}"
done < <(find . -type f -name "*darwin*")
```

Reply:
Reworked to:

```
#!/bin/bash

 if [ -n "${APPLE_CODESIGN_KEY}" ] && [ -n "${APPLE_CODESIGN_PASSWORD}" ] && [ -n "${APPLE_NOTARY_API_KEY}" ]; then
    FILES=$(find -- * -type f -name "*darwin*")
    for file in $FILES; do
      rcodesign sign "${file}" --p12-file="${APPLE_CODESIGN_KEY}" \
        --code-signature-flags=runtime \
        --p12-password-file="${APPLE_CODESIGN_PASSWORD_FILE}"
    done
    zip files.zip "$FILES"
    sha256sum "${ARTIFACTS_TO_PUBLISH}" > checksums.txt
    rcodesign notary-submit files.zip --api-key-path="${APPLE_NOTARY_API_KEY}"  --wait
  fi
```

This is passing shellcheck

Reply:
It works fine

```
root@af5254937cf2:/tmp/latest# FILES=$(find -- * -type f -name "*darwin*")
root@af5254937cf2:/tmp/latest# echo $FILES
kn-darwin-amd64 kn-darwin-arm64
root@af5254937cf2:/tmp/latest# FILES=$(find ./* -type f -name "*darwin*")
root@af5254937cf2:/tmp/latest# echo $FILES
./kn-darwin-amd64 ./kn-darwin-arm64
root@af5254937cf2:/tmp/latest# FILES=$(find * -type f -name "*darwin*")
root@af5254937cf2:/tmp/latest# echo $FILES
kn-darwin-amd64 kn-darwin-arm64
root@af5254937cf2:/tmp/latest# zip files.zip $FILES
updating: kn-darwin-amd64 (deflated 56%)
updating: kn-darwin-arm64 (deflated 57%)
```

Reply:
It works on Linux and Mac

Also `FILES=$(find * -type f -name "*darwin*")`

Can I get lgtm?